src/Microsoft.SqlTools.ServiceLayer/Connection/ReliableConnection/ReliableConnectionHelper.cs

```diff
@@ -46,18 +46,18 @@ internal static class ReliableConnectionHelper
         /// Opens the connection and sets the lock/command timeout and pooling=false.
         /// </summary>
         /// <returns>The opened connection</returns>
-        public static IDbConnection OpenConnection(SqlConnectionStringBuilder csb, bool useRetry)
+        public static IDbConnection OpenConnection(SqlConnectionStringBuilder csb, bool useRetry, string azureAccountToken)
```

Comment:
Should this be an optional parameter?

Reply:
I made it required because everywhere that calls this method needs to take into account the possibility of an azure account token. If it were not required then it would be too easy for us to write code that doesn't use that argument at all and then fails when the user actually has an AAD connection